pkg/resourcegenerator/deployment/deployment.go

```diff
@@ -204,11 +204,16 @@ func Generate(r reconciliation.Reconciliation) error {
 	}
 
 	// add an external link to argocd
-	ingresses := application.Spec.Ingresses
+	hosts, err := application.Spec.Hosts()
+	if err != nil {
+		ctxLog.Error(err, "could not get hosts from application")
```

Comment:
Think we might want to return an error so the reconciliation stops here, and doesn't kill any working applications